Fix config: Marketplace menu item

thomas-kl1 opened the pull request:
Fix config: enable/disable Marketplace menu item

Some code cleanup

thomas-kl1 committed:
Add settings to move only specific modules

thomas-kl1 committed:
fix typo

thomas-kl1 committed:
rever blacklist to whitelist

thomas-kl1 committed:
Fix condition

thomas-kl1 committed:
auto flush admin menu on save

thomas-kl1 committed:
Remove restore default

thomas-kl1 committed:
Config: allow custom tabs

thomas-kl1 committed:
Fix translate labels

thomas-kl1 committed:
add missing delcare strict_types

thomas-kl1 committed:
Introduce allowance rule

thomas-kl1 committed:
Merge branch 'master' of https://github.com/redchamps/clean-admin-menu into redchamps-master

thomas-kl1 committed:
Merge branch 'redchamps-master'

thomas-kl1 committed:
remove unused files

thomas-kl1 committed:
fix settings paths

thomas-kl1 committed:
fix merge issue

thomas-kl1 committed:
fix #5 + code optimization and API/SPI declaration

thomas-kl1 committed:
fix #7 update composer.json

thomas-kl1 committed:
code optim

thomas-kl1 committed:
Add Settings to enable or disable the Magento Extensions menu item

thomas-kl1 committed:
fix magento marketpalce menu item settings

thomas-kl1 committed:
phpcs fixer

thomas-kl1 committed:
Add header license + move marketpalce settings

thomas-kl1 committed:
Fix config path setting

thomas-kl1 committed:
Merge remote-tracking branch 'upstream/master'

thomas-kl1 committed:
remove call_user_func_array callback

thomas-kl1 committed:
Merge branch 'master' of https://github.com/redchamps/clean-admin-menu into redchamps-master

Conflicts:
	Model/AllowedRule/AllowedByDefault.php
	Model/AllowedRule/DeniedByDefault.php
	Model/Config.php
	Model/Config/Source/Rules.php
	Model/IsAllowedStrategy.php
	Model/RuleConfig.php
	Model/RuleFactory.php
	Plugin/Model/ConfigStructure.php
	Plugin/Model/MenuBuilderCommand.php
	etc/adminhtml/di.xml
	etc/adminhtml/system.xml
	etc/config.xml

thomas-kl1 committed:
code cleanup

thomas-kl1 committed:
Merge branch 'redchamps-master'

Conflicts:
	Model/Config.php
	Model/Config/Structure/Data.php
	Model/RuleConfig.php
	Plugin/Model/ConfigStructure.php
	Plugin/Model/MenuBuilderCommand.php
	etc/adminhtml/system.xml
	etc/config.xml

thomas-kl1 committed:
code refactor

thomas-kl1 committed:
remove uneeded dependency

thomas-kl1 committed:
revert wrong changes

thomas-kl1 committed:
fix merge

thomas-kl1 committed:
force config path rewrite

redchamps commented:
Hi @thomas-kl1

Thanks for the pull request.

Does this fix any bug or does it contains code improvements only?

We are working to make extensions vendor setting collapsible so if the pull request doesn’t contain any bug fixes then we would like to mere it along with our change and make a new release.

Please let me know your thoughts.

thomas-kl1 commented:
There is a fix here: etc/adminhtml/menu.xml where the wron config path was set.

The rest of PR is code improvements.

rav-redchamps commented:
Hi @thomas-kl1 

Thanks for sharing this information.

I will be merging this next week along with my change.

rav-redchamps commented on etc/config.xml:
```diff
@@ -11,7 +11,7 @@
                 <allowed_modules><![CDATA[]]></allowed_modules>
                 <rule_modules>deniedByDefault</rule_modules>
-                <allowed_tabs>ddg_automation</allowed_tabs>
+                <allowed_tabs>yotpo,ddg_automation</allowed_tabs>
```
Any reason to allow Yotpo? We have received a request from some people to move this to the Extension section by default.

thomas-kl1 replied:
you are right, this one is something I missed in the merge conflict, my bad

rav-redchamps replied:
No problem. Thanks.

rav-redchamps commented on Model/Config/Structure/Data.php:
```diff
@@ -61,9 +57,8 @@ public function get($path = null, $default = null)
     {
-        $scope = "showIn" . ucfirst($this->scope);
-        return isset($section[$scope]) && $section[$scope];
+        return $section['showIn' . ucfirst($this->_configScope->getCurrentScope())] ?? false;
```
Vendor titles not appearing under Store > Configuration > Extensions. Possibly this change caused this problem.

thomas-kl1 replied:
I don't get the `isSectionVisible` method purpose. I've removed it, and sections not visible in specific scope are not when they should. What is the intent of this method?

rav-redchamps replied:
@thomas-kl1 This method checks if the section is allowed under the currently selected configuration scope global/website/store/store view. so that we add "original tab title"/vendor_title to a visible section only.

This method fixes a  bug that vendor title doesn't appear for some extension vendors having the first section of their tab disallowed for the default scope.

For example, try installing this extension https://www.wyomind.com/magento2/security-enhancement-magento.html and notice Store > Configuration > Extension area. You will not see orange "Wyomind" title for their settings.

thomas-kl1 replied:
I've installed the module and I don't see any issue. The orange title is visible in default, not in website and store, which is normal behavior (there is no available settings in these scopes).
Visibility is already managed Magento core, so to me it's more a duplicate here. Did I missed something in the test?

rav-redchamps replied:
Something might have missed during the test. I will check and get back to you by today or tomorrow.

rav-redchamps replied:
In the meantime, let's not keep the bugfix pending. 

May you please create a separate pull request just for the bug fix or may I just add it from my side and make a release directly. So that it can be merged and we discuss other things after that.

thomas-kl1 replied:
I'll do a separate PR

rav-redchamps replied:
Awesome! Looking forward to it.

rav-redchamps replied:
> I've installed the module and I don't see any issue. The orange title is visible in default, not in website and store, which is normal behavior (there is no available settings in these scopes).
> Visibility is already managed Magento core, so to me it's more a duplicate here. Did I missed something in the test?

Just checked this. It seems like, they have updated their extension. Please try changing file manually to repeat the issue

vendor/wyomind/core/etc/adminhtml/system.xml

Change showInDefault to 0 from 1 for the section tag

thomas-kl1 replied:
> Change showInDefault to 0 from 1 for the section tag

It's definitly because of the third party module bug. I don't think this module should fix the bug of other parties.

rav-redchamps replied:
Yes, it shouldn't. 

But the end-user installing the clean menu extension expects that it should properly move and display all the extension settings under Store > Configuration > Extensions

If a title of extension vendor is not displayed there (due to XYZ reasons), then he will not accept if it's due to the bug in the 3rd party extension instead of clean menu because the title is visible correctly prior installing clean menu.

So, if we can handle such scenario's within our extension then that should be the better approach.

I hope this makes sense.

thomas-kl1 replied:
With the isSectionVisible in Structure Data, when showInDefault is 1 in Wyomind_Core:
![Screenshot from 2020-06-10 09-41-17](https://user-images.githubusercontent.com/20971693/84240616-98313c00-aafe-11ea-948b-cfbe90278f21.png)

isSectionVisible + showInDefault is 0 in Wyomind_Core
![Screenshot from 2020-06-10 09-42-57](https://user-images.githubusercontent.com/20971693/84240785-cca4f800-aafe-11ea-8e81-2c34572e4999.png)
There is no more vendor title even if CronScheduler of Wyomind is there.

Without isSectionVisible + showInDefault is 0 in Wyomind_Core
![Screenshot from 2020-06-10 09-42-57](https://user-images.githubusercontent.com/20971693/84240785-cca4f800-aafe-11ea-8e81-2c34572e4999.png)

Without isSectionVisible + showInDefault is 1 in Wyomind_Core
![Screenshot from 2020-06-10 09-41-17](https://user-images.githubusercontent.com/20971693/84240616-98313c00-aafe-11ea-948b-cfbe90278f21.png)

I don't see any difference by using isSectionVisible (not modified)

So in my opinion there is an other bug, the tab is only depending on the first section visibility.

So I've updated the PR with proper fix

rav-redchamps replied:
Make sense. I am testing the changes and get back to you by today.

rav-redchamps reviewed (changes requested):
Vendor titles not appearing under Store > Configuration > Extensions.

thomas-kl1 committed:
remove isSectionVisible method

peterjaap commented:
Could you change the line in `adminhtml/menu.xml` to;

```
<update id="Magento_Marketplace::partners" dependsOnConfig="clean_admin_menu/settings/enable_marketplace" title="Find Partners &amp; Extensions" resource="Magento_Marketplace::partners" />
```

This not only fixes the dependsOnConfig path, but also makes sure the backend doesn't break when the `magento/module-marketplace` module isn't present (for example, when it is removed through [composer's require directive](https://github.com/yireo/magento2-replace-core/blob/2.3.5/composer.json#L45)).

For now, I'm using a composer patch for [vaimo/composer-patches](https://github.com/vaimo/composer-patches);

```diff
Fix configuration path for marketplace

@package redchamps/module-clean-admin-menu
@version 1.0.5
@link https://github.com/redchamps/clean-admin-menu/pull/10
@level 0


diff --git etc/adminhtml/menu.xml etc/adminhtml/menu.xml
index 9b170bd..cdd444f 100644
--- etc/adminhtml/menu.xml
+++ etc/adminhtml/menu.xml
@@ -8,6 +8,6 @@
 <config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Backend:etc/menu.xsd">
     <menu>
         <add id="RedChamps_CleanMenu::extensions" title="Extensions" translate="title" module="RedChamps_CleanMenu" sortOrder="60" dependsOnModule="RedChamps_CleanMenu" resource="RedChamps_CleanMenu::extensions"/>
-        <update id="Magento_Marketplace::partners" dependsOnConfig="clean_admin_menu/settings/enable_marketplace"/>
+        <update id="Magento_Marketplace::partners" dependsOnConfig="clean_admin_menu/marketplace/enabled" title="Find Partners &amp; Extensions" resource="Magento_Marketplace::partners" />
     </menu>
 </config>
```

thomas-kl1 committed:
add dependsOnModule setting

thomas-kl1 committed:
fix Magento_Marketpalce dependency

thomas-kl1 commented:
@peterjaap You're right, I've also added this attribute: `dependsOnModule="Magento_Marketplace"` and added condition on the module status to show dependent settings in the configuration.

peterjaap commented:
@thomas-kl1 ah great, that's even better.

rav-redchamps commented:
> Could you change the line in `adminhtml/menu.xml` to;
> 
> ```
> <update id="Magento_Marketplace::partners" dependsOnConfig="clean_admin_menu/settings/enable_marketplace" title="Find Partners &amp; Extensions" resource="Magento_Marketplace::partners" />
> ```
> 
> This not only fixes the dependsOnConfig path, but also makes sure the backend doesn't break when the `magento/module-marketplace` module isn't present (for example, when it is removed through [composer's require directive](https://github.com/yireo/magento2-replace-core/blob/2.3.5/composer.json#L45)).
> 
> For now, I'm using a composer patch for [vaimo/composer-patches](https://github.com/vaimo/composer-patches);
> 
> ```diff
> Fix configuration path for marketplace
> 
> @package redchamps/module-clean-admin-menu
> @version 1.0.5
> @link https://github.com/redchamps/clean-admin-menu/pull/10
> @level 0
> 
> 
> diff --git etc/adminhtml/menu.xml etc/adminhtml/menu.xml
> index 9b170bd..cdd444f 100644
> --- etc/adminhtml/menu.xml
> +++ etc/adminhtml/menu.xml
> @@ -8,6 +8,6 @@
>  <config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Backend:etc/menu.xsd">
>      <menu>
>          <add id="RedChamps_CleanMenu::extensions" title="Extensions" translate="title" module="RedChamps_CleanMenu" sortOrder="60" dependsOnModule="RedChamps_CleanMenu" resource="RedChamps_CleanMenu::extensions"/>
> -        <update id="Magento_Marketplace::partners" dependsOnConfig="clean_admin_menu/settings/enable_marketplace"/>
> +        <update id="Magento_Marketplace::partners" dependsOnConfig="clean_admin_menu/marketplace/enabled" title="Find Partners &amp; Extensions" resource="Magento_Marketplace::partners" />
>      </menu>
>  </config>
> ```

@peterjaap Just made a new release 1.0.6 with the fix done by @thomas-kl1 

> @thomas-kl1 ah great, that's even better.

@peterjaap Just made a new release 1.0.6 with the fix done by @thomas-kl1

rav-redchamps commented:
> @thomas-kl1 ah great, that's even better.

@peterjaap Just made a new release 1.0.6 with the fix done by @thomas-kl1

thomas-kl1 committed:
fix tab visibility

rav-redchamps reviewed (approved):
Thanks, again @thomas-kl1 

The changes look stable now. I will be merging the pull request later today.
